velox/functions/sparksql/tests/RegexFunctionsTest.cpp

```diff
@@ -41,6 +43,223 @@ class RegexFunctionsTest : public test::SparkFunctionBaseTest {
     return evaluateOnce<std::string>(
         fmt::format("regexp_extract(c0, '{}')", pattern), str);
   }
+
+  void testRegexReplaceSimpleOnce(
+      const std::vector<std::optional<std::string>>& input,
+      std::string pattern,
+      std::string replace,
+      std::string output,
+      std::optional<int> position =
+          std::nullopt /*adding the optional position parameter*/) {
+    auto valueAt = [&input](vector_size_t row) {
+      return input[row] ? StringView(*input[row]) : StringView();
+    };
+    auto nullAt = [&input](vector_size_t row) {
+      return !input[row].has_value();
+    };
+    auto result = [&] {
+      auto inputString =
+          makeFlatVector<StringView>(input.size(), valueAt, nullAt);
```

Comment:
Can you use `makeFlatVector<std::string>(input)`?

Reply:
Still wonder about this.

Reply:
makeFlatVector<std::string>(input); results in a no matching function call error, but using makeFlatVector<std::string>(input.size(), valueAt, nullAt); compiles and runs with all tests passing.

I changed all cases of makeFlatVector<StringView> to std::string using the second parameter list.

Reply:
makeFlatVector<std::string>(input) or similar should work. Please, take a deeper look. Current code is too verbose.

Reply:
By converting all parameters (except position) to non-optional, the std::string(input) function call worked. This allowed me to remove all valueAt and nullAt lines.

I had to keep position as an optional parameter as it truly is an optional parameter, unlike the others.